pkg/action/action.go

```diff
@@ -538,6 +554,23 @@ func (cfg *Configuration) SetHookOutputFunc(hookOutputFunc func(_, _, _ string)
 	cfg.HookOutputFunc = hookOutputFunc
 }
 
+// Logger returns the logger for the Configuration. If nil, returns slog.Default().
+func (cfg *Configuration) Logger() *slog.Logger {
+	if lg := cfg.logger.Load(); lg != nil {
+		return lg
+	}
+	return slog.Default() // We rarely get here, just be defensive
```

Comment:
It seems not possible for this to be reached, given the logic in `SetLogger()`

That said, why not be consistent with the logic in `SetLogger()`? And return `slog.New(slog.DiscardHandler)` ?

Reply:
> It seems not possible for this to be reached, given the logic in SetLogger()

If `SetLogger()` is never called i.e "do not set a logger", the code can be reached.

```go
	actionConfig := action.NewConfiguration()
	actionConfig.Logger().Info("I'm ready to go!")
```